modules/seqexec/server/src/main/scala/seqexec/server/gmos/Gmos.scala

```diff
@@ -200,6 +215,19 @@ object Gmos {
       ns               <- (if (useNS) nodAndShuffle(config) else NSConfig.NoNodAndShuffle.asRight)
     } yield ns).leftMap(e => SeqexecFailure.Unexpected(ConfigUtilOps.explain(e)))
 
+  def nsConfigF[F[_]: ApplicativeError[?[_], Throwable]](config: Config): F[NSConfig] =
+    ApplicativeError[F, Throwable]
+      .catchNonFatal(
+        nsConfig(config).getOrElse(NSConfig.NoNodAndShuffle)
```

Comment:
I think we chatted about this, but can you confirm that this and `expTime` are deterministic (i.e., we could memoize the results of these method calls)? If not we need to use `Sync` here.

Reply:
So the same `config` param will always return the same result of `nsConfig(config)`